src/localization/making_string_changes.md

```diff
@@ -0,0 +1,117 @@
+In most localization formats, a translation unit consists of a unique identifier (ID, often *string ID* or *message ID*) associated to a segment of text. For example, in the following Fluent snippet, the text `Cancel` is stored in a message with ID `cancel-button`.
+
+```
+cancel-button = Cancel
+```
+
+The same concept applies to legacy formats like DTD or properties, or XML in Android.
+
+```XML
+    # DTD
+    <!ENTITY cancelButton "Cancel">
+
+    # Android XML
+    <string name="cancel_button">Cancel</string>
+```
+
+## General guidelines
+
+These are the general guidelines to follow when updating existing strings already exposed for localization:
+* If you are changing a string such that its **meaning has changed**, you must update the string ID.
+* In case of Fluent, you need to update the string ID also when **the morphology of the message changes**, i.e. attributes are added or removed.
+* If your changes are relevant only for English — for example, to correct a typographical error or to make capitalization consistent — then there is generally no need to update the string ID.
+
+There is a gray area between needing a new ID or not. In some cases, it will be necessary to look at all the existing translations to determine if a new ID would be beneficial. You should always reach out to the l10n team in case of doubt.
+
+As for which new ID to use, versioning – adding a number, e.g. `cancel-button2` – should be only used as a last resource when better IDs are not available.
+
+When updating IDs, make sure to also update references to the old ID in both existing code and comments.
+
+## Why is it necessary to use new IDs?
+
+This is very specific to how localization works at Mozilla, and mostly depends on two key elements:
+* String IDs are used to identify strings throughout the entire localization toolchain.
+* Translations stored in Version Control Systems – Mercurial or GitHub depending on the project – are treated as the **source of truth**.
+
+Changing the message ID will invalidate existing translations, the new message will be reported as missing (new) in all tools, and localizers will have to retranslate it. This is the only reliable method to ensure that localizers update existing localizations, and run-time stops using obsolete translations.
+
+### Generic localized product
+
+Let’s consider the simplest scenario for a localized product, using a single repository for storing both the source locale (en-US) and other localizations.
+
+```
+password-notification = Please enter your master password
+```
+
+This message needs to be updated to `Please enter your primary password`. This is the type of change that we want to be reflected in all locales, so we create a new string, with a different ID:
+
+```
+primary-password-notification = Please enter your primary password
+```
+
+This is effectively a brand new string for the localization toolchain. The downside is that it requires more work for developers to update references in code.
```

Comment:
"The localization toolchain considers this a brand new string and notifies localizers that it requires translation. Pontoon leverages the prior translation, so that the localizer only has to translate the diff (e.g., "Please enter your password" would be leveraged, the localizer would only need to add "primary"). The downside, however, is that it requires more work for developers to update references in code."